packages/shield-controller/src/backend.ts

```diff
@@ -189,44 +197,51 @@ export class ShieldRemoteBackend implements ShieldBackend {
       pollInterval?: number;
     },
   ): Promise<GetCoverageResultResponse> {
+    if (this.#abortController && !this.#abortController.signal.aborted) {
```

Comment:
The way you integrated the abort functionality is problematic.

You use one abort controller for all requests.

There may be situations where you have result polling for different transactions and signatures at the same time. If one log request comes in, it cancels the polling from all other unrelated transactions and signatures. That's most definitely not what we want.